File: service/src/features/feature.go
package features

// COMMON (WEB AND SERVER)

type Feature string

const (
	// Cluster features
	ViewClusterList     Feature = "view.cluster.list"
	ViewClusterItem     Feature = "view.cluster.item"
	ViewClusterOverview Feature = "view.cluster.overview"
	ManageClusterCreate Feature = "manage.cluster.create"
	ManageClusterUpdate Feature = "manage.cluster.update"
	ManageClusterDelete Feature = "manage.cluster.delete"

	// Tags features
	ViewTagList Feature = "view.tag.list"

	// Node features
	ViewNodeDbOverview       Feature = "view.node.db.overview"
	ViewNodeDbConfig         Feature = "view.node.db.config"
	ManageNodeDbConfigUpdate Feature = "manage.node.db.config.update"
	ManageNodeDbSwitchover   Feature = "manage.node.db.switchover"
	ManageNodeDbReinitialize Feature = "manage.node.db.reinitialize"
	ManageNodeDbRestart      Feature = "manage.node.db.restart"
	ManageNodeDbReload       Feature = "manage.node.db.reload"
	ManageNodeDbFailover     Feature = "manage.node.db.failover"
	ManageNodeDbActivation   Feature = "manage.node.db.activation"

	ViewNodeSshMetrics  Feature = "view.node.ssh.metrics"
	ViewNodeSshDocker   Feature = "view.node.ssh.docker"
	ManageNodeSshDocker Feature = "manage.node.ssh.docker"

	// Query features
	ViewQueryCrudList     Feature = "view.query.crud.list"
	ManageQueryCrudCreate Feature = "manage.query.crud.create"
	ManageQueryCrudUpdate Feature = "manage.query.crud.update"
	ManageQueryCrudDelete Feature = "manage.query.crud.delete"

	ViewQueryDbInfo        Feature = "view.query.db.info"
	ViewQueryDbChart       Feature = "view.query.db.chart"
	ManageQueryDbTemplate  Feature = "manage.query.db.template"
	ManageQueryDbConsole   Feature = "manage.query.db.console"
	ManageQueryDbCancel    Feature = "manage.query.db.cancel"
	ManageQueryDbTerminate Feature = "manage.query.db.terminate"

	ViewQueryLogList     Feature = "view.query.log.list"
	ManageQueryLogDelete Feature = "manage.query.log.delete"

	// Vault features
	ViewVaultList     Feature = "view.vault.list"
	ManageVaultCreate Feature = "manage.vault.create"
	ManageVaultUpdate Feature = "manage.vault.update"
	ManageVaultDelete Feature = "manage.vault.delete"

	// Cert features
	ViewCertList     Feature = "view.cert.list"
	ManageCertCreate Feature = "manage.cert.create"
	ManageCertDelete Feature = "manage.cert.delete"

	// Permission management features
	ViewPermissionList     Feature = "view.permission.list"
	ManagePermissionUpdate Feature = "manage.permission.update"
	ManagePermissionDelete Feature = "manage.permission.delete"

	// Tool features
	ViewToolBloatList  Feature = "view.tool.bloat.list"
	ViewToolBloatItem  Feature = "view.tool.bloat.item"
	ViewToolBloatLogs  Feature = "view.tool.bloat.logs"
	ManageToolBloatJob Feature = "manage.tool.bloat.job"

	// Management features
	ViewManagementSecret   Feature = "view.management.secret"
	ManageManagementSecret Feature = "manage.management.secret"
	ManageManagementErase  Feature = "manage.management.erase"
	ManageManagementFree   Feature = "manage.management.free"
	ManageManagementBackup Feature = "manage.management.backup"
)

var All = []Feature{
	ViewClusterList,
	ViewClusterItem,
	ViewClusterOverview,
	ManageClusterCreate,
	ManageClusterUpdate,
	ManageClusterDelete,
	ViewTagList,
	ViewNodeDbOverview,
	ViewNodeDbConfig,
	ViewNodeSshMetrics,
	ManageNodeDbConfigUpdate,
	ManageNodeDbSwitchover,
	ManageNodeDbReinitialize,
	ManageNodeDbRestart,
	ManageNodeDbReload,
	ManageNodeDbFailover,
	ManageNodeDbActivation,
	ManageNodeSshDocker,
	ViewNodeSshDocker,
	ViewQueryCrudList,
	ManageQueryCrudCreate,
	ManageQueryCrudUpdate,
	ManageQueryCrudDelete,
	ViewQueryDbInfo,
	ViewQueryDbChart,
	ManageQueryDbTemplate,
	ManageQueryDbConsole,
	ManageQueryDbCancel,
	ManageQueryDbTerminate,
	ViewQueryLogList,
	ManageQueryLogDelete,
	ViewVaultList,
	ManageVaultCreate,
	ManageVaultUpdate,
	ManageVaultDelete,
	ViewCertList,
	ManageCertCreate,
	ManageCertDelete,
	ViewPermissionList,
	ManagePermissionUpdate,
	ManagePermissionDelete,
	ViewToolBloatList,
	ViewToolBloatItem,
	ViewToolBloatLogs,
	ManageToolBloatJob,
	ViewManagementSecret,
	ManageManagementSecret,
	ManageManagementErase,
	ManageManagementFree,
	ManageManagementBackup,
}

File: web/src/api/feature.ts
// COMMON (WEB AND SERVER)

export enum Feature {
    // Cluster features
    ViewClusterList = "view.cluster.list",
    ViewClusterItem = "view.cluster.item",
    ViewClusterOverview = "view.cluster.overview",
    ManageClusterCreate = "manage.cluster.create",
    ManageClusterUpdate = "manage.cluster.update",
    ManageClusterDelete = "manage.cluster.delete",

    // Tags features
    ViewTagList = "view.tag.list",

    // Node features
    ViewNodeDbOverview = "view.node.db.overview",
    ViewNodeDbConfig = "view.node.db.config",
    ViewNodeSshMetrics = "view.node.ssh.metrics",
    ManageNodeDbConfigUpdate = "manage.node.db.config.update",
    ManageNodeDbSwitchover = "manage.node.db.switchover",
    ManageNodeDbReinitialize = "manage.node.db.reinitialize",
    ManageNodeDbRestart = "manage.node.db.restart",
    ManageNodeDbReload = "manage.node.db.reload",
    ManageNodeDbFailover = "manage.node.db.failover",
    ManageNodeDbActivation = "manage.node.db.activation",
    ManageNodeSshDocker = "manage.node.ssh.docker",
    ViewNodeSshDocker = "view.node.ssh.docker",

    // Query features
    ViewQueryCrudList = "view.query.crud.list",
    ManageQueryCrudCreate = "manage.query.crud.create",
    ManageQueryCrudUpdate = "manage.query.crud.update",
    ManageQueryCrudDelete = "manage.query.crud.delete",
    ViewQueryDbInfo = "view.query.db.info",
    ViewQueryDbChart = "view.query.db.chart",
    ManageQueryDbTemplate = "manage.query.db.template",
    ManageQueryDbConsole = "manage.query.db.console",
    ManageQueryDbCancel = "manage.query.db.cancel",
    ManageQueryDbTerminate = "manage.query.db.terminate",
    ViewQueryLogList = "view.query.log.list",
    ManageQueryLogDelete = "manage.query.log.delete",

    // Vault features
    ViewVaultList = "view.vault.list",
    ManageVaultCreate = "manage.vault.create",
    ManageVaultUpdate = "manage.vault.update",
    ManageVaultDelete = "manage.vault.delete",

    // Cert features
    ViewCertList = "view.cert.list",
    ManageCertCreate = "manage.cert.create",
    ManageCertDelete = "manage.cert.delete",

    // Feature management features
    ViewPermissionList = "view.permission.list",
    ManagePermissionUpdate = "manage.permission.update",
    ManagePermissionDelete = "manage.permission.delete",

    // Bloat features
    ViewToolBloatList = "view.tool.bloat.list",
    ViewToolBloatItem = "view.tool.bloat.item",
    ViewToolBloatLogs = "view.tool.bloat.logs",
    ManageToolBloatJob = "manage.tool.bloat.job",

    // Management features
    ViewManagementSecret = "view.management.secret",
    ManageManagementSecret = "manage.management.secret",
    ManageManagementErase = "manage.management.erase",
    ManageManagementFree = "manage.management.free",
    ManageManagementBackup = "manage.management.backup",
}
